src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -183,11 +200,29 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
         else if (e.OriginalKey() == VirtualKey::Escape)
         {
-            kbdVM.ToggleEditMode();
+            kbdVM.CancelChanges();
             e.Handled(true);
         }
     }
 
+    void Actions::AddNew_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*eventArgs*/)
+    {
+        // Create the new key binding and register all of the event handlers.
+        auto kbdVM{ make_self<KeyBindingViewModel>(_AvailableActionAndArgs) };
+        _RegisterEvents(kbdVM);
+        kbdVM->DeleteNewlyAddedKeyBinding({ this, &Actions::_ViewModelDeleteNewlyAddedKeyBindingHandler });
+
+        // Manually add the editing background. This needs to be done in Actions not the view model.
+        // We also have to do this manually because it hasn't been added to the list yet.
+        kbdVM->IsInEditMode(true);
+        const auto& containerBackground{ Resources().Lookup(box_value(L"ActionContainerBackgroundEditing")).as<Windows::UI::Xaml::Media::Brush>() };
+        kbdVM->ContainerBackground(containerBackground);
+
+        // IMPORTANT: do this _before_ adding the VM to the list. Otherwise, it'll get deleted immediately.

```

Comment:
Why does it get deleted if this property isn't set first? What's the magic? Elaborate?

Reply:
Fixed the comment. I meant to say "after setting IsInEditMode" instead of "before adding to the list".